File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        
        if(root == null) return new ArrayList<Integer>();
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        q.add(root);
        List<Integer> ans = new ArrayList<>();
        
        while(!q.isEmpty()){
            
            int n = q.size();
            int maxi = Integer.MIN_VALUE;
            
            while(n > 0){
                
                TreeNode node = q.element();
                maxi = Math.max(maxi, node.val);
                
                q.poll();
                
                if(node.right != null) q.add(node.right);
                if(node.left != null) q.add(node.left);
                n--;
            }
            ans.add(maxi);
        }
        return ans;
    }
}

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        
        if(!root) return {};
        queue<TreeNode*> q;
        q.push(root);
        vector<int> arr;
        
        while(!q.empty()){
            
            int n = q.size();
            long long maxi = -1e10;
            
            while(n--){
                
                TreeNode* node = q.front();
                maxi = max(maxi, (long long)node->val); 
                q.pop();
                
                if(node->right) q.push(node->right);
                if(node->left) q.push(node->left);
            }
            arr.push_back(maxi);
        }
        return arr;
    }
};